docs/designs/dynamic_sharding_for_images.md

```diff
@@ -0,0 +1,207 @@
+# Dynamic Sharding Support Reading Original Images
+
+Now, users have to convert images into RecordIO for ElasticDL. In practice,
+many users read the original images to train their models. So, we need to
+support read the original data from the storage. However, there are various
+file formats for the original data. We need to design a common definition
+for different formats.
+
+## Different Ways to Store Images and Annotations
+
+1. All images are in the same folder.
+
+    ```txt
+    |-- images
+        |--0001.png
+        |--0002.png
+        |--0003.png
+        |--0004.png
+        |--0005.png
+    ```
+
+    In the format, users may not need labels in the image compression.
```

Comment:
For this case, how can we get the label data for training?

Reply:
Sometimes, users don't need the label such as the CNN encoder model.